ebi_eva_common_pyutils/nextflow/linear_pipeline.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2021 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from ebi_eva_common_pyutils.logger import AppLogger
+
+
+class LinearNextFlowPipeline(AppLogger):
```

Comment:
I'm really not sure of what the will be the benefit of generating Nextflow pipeline from python.
As soon as we're going to want to do anything slightly more complicated than execute couple of commands this interface is not going to work.
Happy to discuss this though.

Reply:
Yes, this being the "linear" pipeline, it can only execute a series of commands but this comes in handy if we have to execute the same set of steps for a set of species. We did a similar thing for release pipeline and now the same use-case is there for mongo migration (where we execute same set of commands: dump, prepare, restore, index for a set of species).

Reply:
I can see the case for what Sundar mentions although I wonder if this is something we would continue using as I think  that should be the main reason to decide if something should be included in the eva-common-pyutils library

The automation is definitely needed for moving the databases as we will be running this for every database in our mongo cluster and I see 3 options.
1. Doing it with an automation script written in python just as the release process
2. Generating dynamically the nextflow pipeline from python (as it is currently implemented)
3. Writing the process as a nextflow pipeline and wrap it with a python script to provide the necessary parameters for each database

I guess we can discuss how we want to do it as I would like to be consistent in the way we automate moving forward

Reply:
We have many database remediation/data correction processes coming up in the "Internal data integrity" epic. I felt that some of those processes would be well-served by creating simple linear pipelines. Since the same logic might surface across multiple projects, I thought I could refactor the logic into a simple class (as dumb as its functionality is :) ).